EIPS/eip-7792.md

```diff
@@ -0,0 +1,126 @@
+---
+eip: 7792
+title: Verifiable logs
+description: Scheme to make the eth_getLogs response verifiable
+author: Etan Kissling (@etan-status), Gajinder Singh (@g11tech), Vitalik Buterin (@vbuterin)
+discussions-to: https://ethereum-magicians.org/t/eip-7792-verifiable-logs/21424
+status: Draft
+type: Standards Track
+category: Core
+created: 2024-10-21
+requires: 6466
+---
+
+## Abstract
+
+This EIP defines a method to make the `eth_getLogs` JSON-RPC response verifiable.
+
+## Motivation
+
+The `eth_getLogs` endpoint is used by wallets to obtain the transaction history pertaining to an account or a topic. To verify correctness and completeness of the logs, a wallet would also have to obtain all block headers and check against their logs bloom. However, that mechanism is inefficient due to its high false positive rate and also involves an unpractical amount of network round trips. This EIP defines a replacement mechanism to efficiently and incrementally verify correctness and completeness of `eth_getLogs` responses.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Configuration
+
+| Name | Value |
+| - | - |
+| `LOG_CONTRACT_ADDRESS` | `0xfffffffffffffffffffffffffffffffffffffffe` |
+
+### Log accumulation
+
+After executing all transactions of a block, commitments of all emitted logs are accumulated into the storage of `LOG_CONTRACT_ADDRESS`. The contract has no code, and its storage layout consists of three slots of type `mapping`. However to prevent [EIP-158](./eip-158.md) cleanup, the contract's nonce is set to `1` at the first write.
+
+| Name | Value | Type |
+| `LOG_ADDRESS_STORAGE_SLOT` | `0` | `mapping(address => bytes32)` |
+| `LOG_TOPICS_STORAGE_SLOT` | `1` | `mapping(bytes32 => bytes32)` |
+| `LOG_ADDRESS_TOPICS_STORAGE_SLOT` | `2` | `mapping(bytes32 => bytes32)` |
```

Comment:
This looks like a solidity-like type and should include how to decode the mapping to get the actual storage key needed here.